prime-router/src/main/kotlin/Report.kt

```diff
@@ -1121,23 +1130,97 @@ class Report : Logging {
         return StringColumn.create(name, List(itemCount) { "" })
     }
 
-    private fun buildRestritedZipCode(name: String): StringColumn {
-        val restricted_zip = metadata.findLookupTable("restricted_zip_code")
-        var row = 0
+    /**
+     * Given a column name, this function walks through each value and if the value in that
+     * column matches a restricted postal code, it will replace it with the appropriate value
+     * per the HIPAA Safe Harbor rules
+     * @param name The name of the column to examine
+     */
+    private fun buildRestrictedZipCode(name: String): StringColumn {
+        val restrictedZip = metadata.findLookupTable("restricted_zip_code")
 
-        table.column(name).forEach {
+        table.column(name).forEachIndexed { idx, columnValue ->
             // Assuming zip format is xxxxx-yyyy
-            val zipCode = it.toString().split("-")
+            val zipCode = columnValue.toString().split("-")
             val value = zipCode[0].dropLast(2)
-            if (restricted_zip?.dataRows?.contains(listOf(value)) == true) {
-                setString(row++, name, "00000")
+            if (restrictedZip?.dataRows?.contains(listOf(value)) == true) {
+                setString(idx, name, "00000")
             } else {
-                setString(row++, name, (value + "00"))
+                setString(idx, name, (value + "00"))
             }
         }
         return table.column(name).copy() as StringColumn
     }
 
+    /**
+     * Walks the table rows and compares the patient age and if it is greater than or
+     * equal to the comparison value, it will zero it out, otherwise it will pass it through
+     * unchanged.
+     */
+    private fun buildDeidentifiedPatientAgeColumn(nullValuePlaceholder: String = ""): StringColumn {
+        // loop through the table rows
+        table.forEachIndexed { idx, row ->
+            // get the specimen collection date
+            val specimenCollectionDateTime = row
+                // get the specimen collection date time
+                .getStringOrNull(specimen_collection_date_column_name)
+                // trim it just in case
+                .trimToNull()
+                .let {
+                    when (it) {
+                        // if the value is not null, parse it to a date value, otherwise, use current
+                        // date time value to compare DOB against
+                        null -> DateUtilities.nowAtZone(DateUtilities.utcZone).toOffsetDateTime(DateUtilities.utcZone)
+                        else -> DateUtilities.parseDate(it).toOffsetDateTime(DateUtilities.utcZone)
+                    }
+                }
+            // get the patient age
+            val patientAge = getAge(
+                row.getStringOrNull(patient_age_column_name).trimToNull(),
```

Comment:
Seems like `getStringOrNull()` already calls `trimToNull()`?

```kotlin
  private fun Row.getStringOrNull(columnName: String): String? {
      return this.getStringOrDefault(columnName, null).trimToNull()
  }
```

(applies to multiple places in this PR)